restcomm/restcomm.dao/src/main/java/org/restcomm/connect/dao/mybatis/MybatisIncomingPhoneNumbersDao.java

```diff
@@ -70,19 +74,61 @@ public IncomingPhoneNumber getIncomingPhoneNumber(final String phoneNumber) {
         return getIncomingPhoneNumber("getIncomingPhoneNumberByValue", phoneNumber);
     }
 
-    private IncomingPhoneNumber getIncomingPhoneNumber(final String selector, Object parameter) {
+   private IncomingPhoneNumber getIncomingPhoneNumber(final String selector, Object parameter) {
         final SqlSession session = sessions.openSession();
-        try {
+        String inboundPhoneNumber = null;
+          try {
+            //if(parameter.toString().contains("*")){}{
+            //perform direct check
             final Map<String, Object> result = session.selectOne(namespace + selector, parameter);
-            if (result != null) {
+            if (result != null ) {
                 return toIncomingPhoneNumber(result);
-            } else {
-                return null;
-            }
-        } finally {
+            }//}
+            //check if there is a Regex match only if parameter is a String aka phone Number
+            if(!(parameter instanceof Sid)){
+               inboundPhoneNumber = parameter.toString().replace("+1", "");
+               logger.warn(" Inbound phoneNumber : " + inboundPhoneNumber );
+              return checkIncomingPhoneNumberRegexMatch(selector, inboundPhoneNumber);
+               }
+
+        }finally {
+                     session.close();
+         }
+          return null;
+
+}
+
+   public IncomingPhoneNumber checkIncomingPhoneNumberRegexMatch ( String selector, String inBoundPhoneNumber){
+               final SqlSession session = sessions.openSession();
+               String phoneRegexPattern = null;
+               try {
+                    List<IncomingPhoneNumber> listPhones = getAllIncomingPhoneNumbers();
+                   for (IncomingPhoneNumber listPhone : listPhones){
```

Comment:
@gvagenas, that is true but I didn't want to mess with the exact match we already have in place. Another thing is that the exact match is done using predefined SQL that we have defined in the https://github.com/RestComm/Restcomm-Connect/blob/master/restcomm/restcomm.application/src/main/webapp/WEB-INF/sql/incoming-phone-numbers.xml

The other option would be the parse the Regex using mysql but that might be problematic if mysql or mariadb version changed as we have seen when we do DB upgrade.